### PR TITLE
Java 21 and spring boot migration changes with terraform changes

### PR DESCRIPTION
**Pull request by dgurjar-ch**
__Short description outlining key changes/additions__ : java 21 and spring boot 3 migration with aws terraform  changes 

__JIRA Ticket Number__: JU-1146

### Type of change

* [x] New feature

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
-documentgeneration.baseUrl=/private/documents/generate
\ No newline at end of file
+documentgeneration.baseUrl=/private/documents/generate
+
+
+management.endpoints.web.base-path=/
+management.endpoints.web.path-mapping.health=document-generator-consumer/healthcheck
+management.endpoint.health.show-details=never
+management.endpoint.health.enabled=true
+management.health.db.enabled=false
```
Comment (dhawes-ch) [inline src/main/resources/application.properties:10]: [You should explicitly set the default access to disabled](https://docs.spring.io/spring-boot/reference/actuator/endpoints.html#actuator.endpoints.controlling-access), so we're certain we're only exposing the health check

File: src/main/java/uk/gov/companieshouse/document/generator/consumer/interceptor/LoggingInterceptor.java
Comment (dhawes-ch) [inline src/main/java/uk/gov/companieshouse/document/generator/consumer/interceptor/LoggingInterceptor.java:1]: The postHandle on these interceptors has weird behaviour, it doesn't produce log output until a new request comes in, at which point it prints the end of the previous request, then the start of the new request, back to back. So if there's a delay between requests you will always have what looks like a partially incomplete request.

For now you can get around this by adding a trace log after the end request log, e.g https://github.com/companieshouse/chs-monitor-api-java/blob/1173244d20115f254322780acb67feffec3c0acb/src/main/java/uk/gov/companieshouse/chsmonitorapi/logging/RequestLogInterceptor.java#L35

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by foman-ch — APPROVED** [review]

**Pull request merged**